### ISSUE TITLE
minminguard 1.9.1 crash

### ISSUE THREAD

**Issue by kataki** [chiehmin/MinMinGuard#29]
whenever i try to run minminguard 1.9.0 and 1.9.1 its crashing, i tried clearing data, uninstalling and installing, disabling all other modules in xposed but no luck at all 
ver 1.9.0 was working fine until one day i noticed it started crashing and now 1.9.1 is doing the same
i uploaded logcat if you can check plz
https://www.dropbox.com/s/afkc4mlumhhq7lq/logcat_and_device_info.zip?dl=0
thanks & regards

**Comment by chiehmin** [comment]
Thanks for report and logcat:)

**Comment by pylerSM** [comment]
Did you try clearing app data?

**Comment by pylerSM** [comment]
Ok, you tried. Looks like app can't retrieve icon for some system app but that is not problem since we get default system one.

The strange thing is OutOfMemory error. Maybe we could add option to enable/disable icon loading if it used so much memory.

**Comment by chiehmin** [comment]
I think we can just surround the getIcon fuction with try and catch. Leave the icon blank if  we can not retrieve it.

**Comment by kataki** [comment]
could be because of number of apps installed is too much around 800 apps?

**Comment by chiehmin** [comment]
This may be the reason

**Comment by pylerSM** [comment]
So we should disable icons loading if number of apps is too high.

**Comment by pylerSM** [comment]
I tried to do something. https://github.com/chiehmin/MinMinGuard/pull/36 Please build test APK and verify it whether it still crashes.

**Comment by pylerSM** [comment]
Or maybe disable icon loading if show system apps is enabled (since this option increases number of apps in apps list)

**Comment by pylerSM** [comment]
People with 700-800 apps get crashes. So I would try limit 500 apps.

**Comment by chiehmin** [comment]
This is a similar question on stackoverflow.

http://stackoverflow.com/questions/16958879/getting-a-lot-of-app-icons-make-out-of-memory

**Comment by pylerSM** [comment]
Yes, but I am not sure what we exactly need that way. It would require some more changes and and I don't think we can live without icons (when users has 500 and more apps). Search is still there. 

So I prefer simple, elegant solution.
Idea: if showSystemApps is enabled and number of apps (user+system) > 500, don't load app icons. I will adjust commit with this logic.

**Comment by pylerSM** [comment]
Done

**Comment by kataki** [comment]
I tried v2.0.0 its still crashing
log
https://www.dropbox.com/s/afkc4mlumhhq7lq/logcat_and_device_info.zip?dl=0

**Comment by pylerSM** [comment]
Because fix was not merged yet.

**Comment by kataki** [comment]
thanks, any idea when the fix will be merged?

**Comment by pylerSM** [comment]
When FatMinMin will have time for it.

**Comment by chiehmin** [comment]
Should be fixed by commit a7554f1

**Comment by NitroBAY** [comment]
Same here even though with 2.0. But I'm not here for that but to express my happiness because I find somebody else who have +600 apps. When I used buzz launcher I sent an email to complain about the fact that the folder where limited to 100 and so I had to create 3 for my games. They say that I was the first one who find this limit. So I DL nova which even more famous than buzz (tough buzz have +10M) but it crash when I started to put app in a folder. I felt in love with action launcher and buy the pro but starting from 50 apps in a single folder the launcher crash and lag if I put outside. I reported via email 4month ago but still buggy and on Google plus few days before but it seemed that I'm the only one to have problem with launcher. And even worst on action launcher no one of millions of users seemed to have +50 apps. I search for a long time people like me, thanks kataki
